File: backend/app/application/organization/commands/create_organization.py
"""
Command: create organization.
"""

from dataclasses import dataclass


@dataclass(frozen=True)
class CreateOrganizationCommand:
    """Create organization command."""

    name: str
    short_name: str | None = None
    inn: str | None = None
    kpp: str | None = None
    ogrn: str | None = None
    address: str | None = None
    phone: str | None = None
    email: str | None = None
    website: str | None = None
    comment: str | None = None

File: frontend/src/features/organizations/api/createOrganization.ts
import { http } from "../../../shared/api/http";
import type { OrganizationRead } from "../model/types";
import { mapOrganization } from "./organizationMapper";
import type { OrganizationApiDto } from "./types";
import type { CreateOrganizationForm } from "../model/types";

export async function createOrganization(
  data: CreateOrganizationForm,
): Promise<OrganizationRead> {
  const response = await http.post<OrganizationApiDto>("/organizations/", {
    name: data.name,
    short_name: data.shortName?.trim() || null,
    inn: data.inn?.trim() || null,
    kpp: data.kpp?.trim() || null,
    ogrn: data.ogrn?.trim() || null,
    address: data.address?.trim() || null,
    phone: data.phone?.trim() || null,
    email: data.email?.trim() || null,
    website: data.website?.trim() || null,
    comment: data.comment?.trim() || null,
  });

  return mapOrganization(response.data);
}
